trading/trader/modes/abstract_mode_creator.py

```diff
@@ -64,64 +75,105 @@ def get_additional_dusts_to_quantity_if_necessary(quantity, price, symbol_market
+
+                    # invalid order
+                    return []
+
+                # check total_order_price not > max_cost and valid_quantity not > max_quantity
+                elif total_order_price > max_cost or valid_quantity > max_quantity:
+                    # split quantity into smaller orders
+                    nb_orders_according_to_cost = total_order_price / max_cost
+                    nb_orders_according_to_quantity = valid_quantity / max_quantity
+                    if nb_orders_according_to_cost > nb_orders_according_to_quantity:
+                        return AbstractTradingModeCreator._adapt_order_quantity_because_price(total_order_price,
+                                                                                              max_cost, price,
+                                                                                              symbol_market)
+                    else:
+                        return AbstractTradingModeCreator\
+                            ._adapt_order_quantity_because_quantity(valid_quantity, max_quantity,
+                                                                    quantity, price, symbol_market)
+
+                else:
+                    # valid order that can be handled wy the exchange
+                    return [(valid_quantity, valid_price)]
+
+            # case 1.2: use only quantity and price
+            elif self._is_valid(limit_price, Ecmsc.LIMITS_PRICE_MIN.value) \
+                    and self._is_valid(limit_price, Ecmsc.LIMITS_PRICE_MAX.value):
+
+                min_price = AbstractTradingModeCreator.get_value_or_default(limit_price, Ecmsc.LIMITS_PRICE_MIN.value)
+                max_price = AbstractTradingModeCreator.get_value_or_default(limit_price, Ecmsc.LIMITS_PRICE_MAX.value)
+
+                if not (max_price >= valid_price >= min_price):
+                    # invalid order
+                    return []
+
+                # check total_order_price not > max_cost and valid_quantity not > max_quantity
+                elif valid_quantity > max_quantity:
+                    # split quantity into smaller orders
+                    return AbstractTradingModeCreator \
+                        ._adapt_order_quantity_because_quantity(valid_quantity, max_quantity,
+                                                                quantity, price, symbol_market)
+                else:
+                    # valid order that can be handled wy the exchange
+                    return [(valid_quantity, valid_price)]
+
+        if not fixed_symbol_data:
+            # case 2: try fixing data from exchanges
+            fixed_symbol_data = ExchangeMarketStatusFixer(symbol_market, price).get_market_status()
+            return self.check_and_adapt_order_details_if_necessary(quantity, price,
+                                                                   fixed_symbol_data, fixed_symbol_data=True)
```

Comment:
Can you split this code / function ? (for clarity)

Reply:
done